src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -2723,15 +2722,11 @@ trait Types
             case xs =>
               xs.init.mkString("(", ", ", ")") + " => " + xs.last
           }
-        }
-        else if (isShowAsInfixType)
-          infixTypeString
-        else if (isTupleTypeDirect(this))
-          tupleTypeString
-        else if (sym.isAliasType && prefixChain.exists(_.termSymbol.isSynthetic) && (this ne dealias))
-          "" + dealias
-        else
-          ""
+        case _ if isShowAsInfixType                    => infixTypeString
+        case _ if isTupleTypeDirect(this)              => tupleTypeString
+        case _ if sym.isAliasType && prefixChain.exists(_.termSymbol.isSynthetic) && (this ne dealias)
```

Comment:
```suggestion
        case _ if sym.isAliasType && (this ne dealias) && prefixChain.exists(_.termSymbol.isSynthetic)
```

Reply:
It's not just readability, it's also bringing a quicker condition first and short-cutting the `exists`.